$OpenBSD: patch-src_3rdparty_javascriptcore_JavaScriptCore_runtime_JSValue_h,v 1.2 2010/10/28 14:55:47 espie Exp $
Fix build on sparc64, from netbsd x11/qt4-libs/patches/path-au
--- src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSValue.h.orig	Fri Sep 10 11:05:24 2010
+++ src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSValue.h	Sun Oct 24 14:16:41 2010
@@ -491,7 +491,7 @@ namespace JSC {
             u.asBits.tag = CellTag;
         else
             u.asBits.tag = EmptyValueTag;
-        u.asBits.payload = reinterpret_cast<int32_t>(ptr);
+        u.asBits.payload = reinterpret_cast<intptr_t>(ptr);
 #if ENABLE(JSC_ZOMBIES)
         ASSERT(!isZombie());
 #endif
@@ -503,7 +503,7 @@ namespace JSC {
             u.asBits.tag = CellTag;
         else
             u.asBits.tag = EmptyValueTag;
-        u.asBits.payload = reinterpret_cast<int32_t>(const_cast<JSCell*>(ptr));
+        u.asBits.payload = reinterpret_cast<intptr_t>(const_cast<JSCell*>(ptr));
 #if ENABLE(JSC_ZOMBIES)
         ASSERT(!isZombie());
 #endif
